### ISSUE TITLE
Feature Request: Deletion of entities via specification

### ISSUE THREAD

**Issue by thorstenalpers** [ardalis/Specification#363]
Hi, Ardalis Specification is a really great project that I use extensively as Nuget. I have an additional feature request. I have implemented in my own EfRepository, but would like to see in your project. It is about deleting entities per specification.

The existing way is to list entities and then delete them.

```
 public abstract class RepositoryBase<T> : IReadRepository<T> where T : class
{
...
   public virtual async Task DeleteRangeAsync(ISpecification<T> specification, CancellationToken cancellationToken = default)
    {
        var query = ApplySpecification(specification);
        _dbContext.Set<T>().RemoveRange(query);
        await SaveChangesAsync(cancellationToken);
    }
...
}
```

I could also create a PR with unittests.

**Comment by fiseni** [comment]
Hi @thorstenalpers 

Yea we may add it. Do you want to give it a try?
@ardalis do you have any objections?

**Comment by thorstenalpers** [comment]
Yeah, would be great!

**Comment by ardalis** [comment]
No objections, go for it @thorstenalpers !

**Comment by thorstenalpers** [comment]
@ardalis I have created a fork and created a branch "feature/delete-range-by-spec" and created a PR in my repository. 

https://github.com/thorstenalpers/Specification/tree/talpers/delete-range-by-spec

Do you want to sync the fork, so that you have the feature branch and you are able to merge it into main?

**Comment by fiseni** [comment]
Resolved by PR #369
